evo_researcher/functions/search.py

```diff
@@ -1,14 +1,24 @@
+import requests
 from evo_researcher.functions.web_search import WebSearchResult, web_search
 from concurrent.futures import ThreadPoolExecutor, as_completed
 
-def search(queries: list[str], api_key: str, filter = lambda x: True) -> list[tuple[str, WebSearchResult]]:
+
+def safe_web_search(query: str, max_results=5) -> list[WebSearchResult]:
+    try:
+        return web_search(query, max_results)
+    except requests.exceptions.HTTPError as e:
+        print(f"Error in web_search: {e}")
+        return []
```

Comment:
No need to ruin the whole prediction because of some errored scrape.

Reply:
I don't feel great about this. Without inspecting the results of the search step you won't know if your agent makes a bad prediction because it only got 2 urls to scrape from, as apposed to something else.

I know you've already tried to fix the falkiness of the API with `@tenacity.retry`, but I think if we go with this then we at least need to have visibility of "tried to search for 20 urls, only got 4".

Reply:
Added a print warning, later we should replace all the prints with propper logging 😄

Reply:
Yes. It would also be a good feature for the benchmarker to capture the logging for each agent and append the warnings to the report